wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -14,39 +18,65 @@ module Cardano.Wallet.WalletLayer.Types
     , updateAccount
     , deleteAccount
 
+    , createAddress
     , getAddresses
+    , isAddressValid
+
+    , createTx
+    , getTxs
+    , estimateFees
+
     ) where
 
+
 import           Universum
 
+import           Control.Exception.Safe (try)
 import           Control.Lens (makeLenses)
 
-import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, Address,
-                                              NewAccount, NewWallet, Wallet, WalletId, WalletUpdate)
+import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, AddressValidity,
+                                              EstimatedFees, NewAccount, NewAddress, NewWallet,
+                                              Payment, Transaction, V1, Wallet, WalletAddress,
+                                              WalletId, WalletUpdate)
+import           Pos.Core (Address, TxAux)
 
 import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
+import           Cardano.Wallet.WalletLayer.Error (WalletLayerError (..))
 
 ------------------------------------------------------------
 -- Passive wallet layer
 ------------------------------------------------------------
 
+-- The response for the @WalletPassiveLayer@.
+type WalletLayerResponse m a = m (Either WalletLayerError a)
```

Comment:
I know that this is how the HTTP client does things, but the reason I chose that representation is because of the HTTP-nature of the program -- HTTP stuff can always fail (eg with a `ServantErr`), so it makes sense to wrap all of the endpoints with that. However, the `WalletLayer` is operating locally on a much more trusted system -- the local database. Because of that, we should *expect* that a catastrophic, transport-level failure is much more rare, and we can use things like "The database disappeared off of the disk" as an exceptional situation.

For the wallet layer, I would prefer to see the possible failures for an action represented in the return type of the action. If getting wallets has an error, then the return of `pwlGetWallets` should have `Either GetWalletError Wallet`. Same for accounts, etc. 

This would mean breaking up the `WalletLayerError` into smaller, finer types, and returning those from relevant functions instead of the monolithic error type.